flair/models/one_class_classification_model.py

```diff
@@ -0,0 +1,226 @@
+    distinguishing between two or more classes.
+
+    Example:
+    -------
+    The model expects to be trained on a dataset in which every element has the same label_value, e.g. movie reviews
+    with the label POSITIVE.
+    During inference, one of two label_values will be added:
+    - In-class (e.g. another movie review) -> label_value="POSITIVE"
+    - Anything else (e.g. a wiki page)     -> label_value="<unk>"
+
+    Architecture
+    ------------
+    Reconstruction with autoencoder. The score is the reconstruction error from compressing and decompressing the
+    document embedding. A LOWER score indicates a HIGHER probability of being in-class. The threshold is
+    calculated as a high percentile of the score distribution of in-class elements from the dev set.
+
+    You must set the threshold after training by running `model.threshold = model.calculate_threshold(corpus.dev)`.
+    """
+
+    def __init__(
+        self,
+        embeddings: DocumentEmbeddings,
+        label_dictionary: Dictionary,
+        label_type: str,
+        encoding_dim: int = 128,
+        threshold: Optional[float] = None,
+    ) -> None:
+        """Initializes a OneClassClassifier.
+
+        Args:
+        embeddings: Embeddings to use during training and prediction
+        label_dictionary: The label to predict. Must contain exactly one class.
+        label_type: name of the annotation_layer to be predicted in case a corpus has multiple annotations
+        encoding_dim: The size of the compressed embedding
+        threshold: The score that separates in-class from out-of-class
+        """
+        super().__init__()
+        self.embeddings = embeddings
+        if len(label_dictionary) != 1:
+            raise ValueError(f"label_dictionary must have exactly 1 element: {label_dictionary}")
+        self.label_dictionary = label_dictionary
+        self.label_value = label_dictionary.get_items()[0]
+        self._label_type = label_type
+        self.encoding_dim = encoding_dim
+        self.threshold = threshold
+
+        embedding_dim = embeddings.embedding_length
+        self.encoder = torch.nn.Sequential(
+            torch.nn.Linear(embedding_dim, encoding_dim * 4),
+            torch.nn.LeakyReLU(True),
+            torch.nn.Linear(encoding_dim * 4, encoding_dim * 2),
+            torch.nn.LeakyReLU(True),
+            torch.nn.Linear(encoding_dim * 2, encoding_dim),
+            torch.nn.LeakyReLU(True),
+        )
+
+        self.decoder = torch.nn.Sequential(
+            torch.nn.Linear(encoding_dim, encoding_dim * 2),
+            torch.nn.LeakyReLU(True),
+            torch.nn.Linear(encoding_dim * 2, encoding_dim * 4),
+            torch.nn.LeakyReLU(True),
+            torch.nn.Linear(encoding_dim * 4, embedding_dim),
+            torch.nn.LeakyReLU(True),
+        )
+
+        self.cosine_sim = torch.nn.CosineSimilarity(dim=1)
```

Comment:
Out of curiosity, did you try out/look into other loss functions?

Reply:
I haven't done anything too rigorous, but I did take a quick look with euclidean. It didn't seem to perform as well so I hardcoded cosine distance, but I see some other models have a parameter to allow using different losses. I'm happy to add that in if you'd like.

For example, running the script in the top comment with `OneClassClassifier` edited to use euclidean loss, I get f-scores around 0.8 with more variance (f-score will still go up to 1 as the training size is increased). As written with cosine, it gives f-scores around 1.0.

Reply:
Thanks, I think leaving it just with cosine is good for now!